bqskit/ir/opt/instantiaters/qfactor.py

```diff
@@ -22,151 +20,13 @@
-            )
-
-        if diff_tol_a <= 0 or diff_tol_a >= 0.5:
-            raise ValueError(
-                'Expected 0 < diff_tol_a < 0.5, got %d.' % diff_tol_a,
-            )
-
-        if not is_real_number(diff_tol_r):
-            raise TypeError(
-                'Expected float for diff_tol_r, got %s.' % type(diff_tol_r),
-            )
-
-        if diff_tol_r <= 0 or diff_tol_r >= 0.5:
-            raise ValueError(
-                'Expected 0 < diff_tol_r < 0.5, got %d.' % diff_tol_r,
-            )
-
-        if not is_real_number(dist_tol):
-            raise TypeError(
-                'Expected float for dist_tol, got %s.' % type(dist_tol),
-            )
-
-        if dist_tol <= 0 or dist_tol >= 1:
-            raise ValueError(
-                'Expected 0 < dist_tol < 1, got %d.' % dist_tol,
-            )
-
-        if not is_integer(max_iters):
-            raise TypeError(
-                'Expected int for max_iters, got %s.' % type(max_iters),
-            )
-
-        if max_iters < 0:
-            raise ValueError(
-                'Expected positive integer for max_iters, got %d.' % max_iters,
-            )
-
-        if not is_integer(min_iters):
-            raise TypeError(
-                'Expected int for min_iters, got %s.' % type(min_iters),
-            )
-
-        if min_iters < 0:
-            raise ValueError(
-                'Expected nonnegative integer for min_iters'
-                ', got %d.' % min_iters,
-            )
-
-        if not is_real_number(slowdown_factor):
-            raise TypeError(
-                'Expected float for slowdown_factor, got %s.'
-                % type(slowdown_factor),
-            )
-
-        if slowdown_factor < 0 or slowdown_factor >= 1:
-            raise ValueError(
-                'Expected 0 <= slowdown_factor < 1, got %d.' % slowdown_factor,
-            )
-
-        if not is_integer(reinit_delay):
-            raise TypeError(
-                'Expected int for reinit_delay, got %s.' % type(reinit_delay),
-            )
-
-        if reinit_delay <= 0:
-            raise ValueError(
-                'Expected nonnegative integer for reinit_delay'
-                ', got %d.' % reinit_delay,
-            )
-
-        if not is_integer(log_delay):
-            raise TypeError(
-                'Expected int for log_delay, got %s.' % type(log_delay),
-            )
-
-        if log_delay <= 0:
-            raise ValueError(
-                'Expected nonnegative integer for log_delay'
-                ', got %d.' % log_delay,
-            )
-
-        self.diff_tol_a = diff_tol_a
-        self.diff_tol_r = diff_tol_r
-        self.dist_tol = dist_tol
-        self.max_iters = max_iters
-        self.min_iters = min_iters
-        self.slowdown_factor = slowdown_factor
-        self.reinit_delay = reinit_delay
-        self.log_delay = log_delay
+    def __new__(cls, **kwargs: dict[str, Any]) -> Any:
```

Comment:
Is there a way to handle at a higher level, like in `circuit.minimize`?

Reply:
Probably, I think we should still have the `__new__` but it should (at least for now) reject all non-empty kwargs.